### PR TITLE
Cisco SB - Bugfix for 16544

### PR DESCRIPTION
**Pull request by rudybroersma**
This PR fixes Cisco SB discovery for issue #16544
Also moved this to SnmpQuery

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [ ] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [ ] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.
- [ ] If my Pull Request makes discovery/polling/yaml changes, I have added/updated [test data](https://docs.librenms.org/Developing/os/Test-Units/).

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Comment by laf** [comment]
This needs a rebase @rudybroersma, also is it still supposed to be draft?

**Comment by laf** [comment]
This looks to be the same PR as #16776, which one is the right one?

**Comment by rudybroersma** [comment]
> This looks to be the same PR as #16776, which one is the right one?

It's not the same. 16776 is for Cisco NXOS. This one is for Cisco Small Business

**Comment by rudybroersma** [comment]
What's going on with the tests? I've had the tests fail on the other PR as well with the same issues:

-    'links.8.remote_port' => 'te1/0/1'
+    'links.8.remote_port' => 'te1/0/1 (c4c60388384f)'

What's up with the hexadecimal port identifier?

**Comment by rudybroersma** [comment]
I seem to be having a lot of issues with test data lately. Not sure why my installation is getting wildly different test results compared to the CI/CD tests

File: tests/data/ciscosb_cbs350-4x.json [hunk 1]
```diff
@@ -12058,7 +12058,7 @@
                     "sensor_oid": ".1.3.6.1.4.1.9.6.1.101.108.1.1.5.1.10",
                     "sensor_index": "1.10",
                     "sensor_type": "ciscosb",
-                    "sensor_descr": "GigabitEthernet1/0/10 PoE",
+                    "sensor_descr": "PoE",
```
Comment (PipoCanaja) [inline tests/data/ciscosb_cbs350-4x.json:12061]: Hi @rudybroersma 
Seems that the descr is now "PoE" for all sensors. Could you fix that ?

Comment (rudybroersma) [reply tests/data/ciscosb_cbs350-4x.json:12061]: Problem is, I haven't changed that.  I'm only reading "CISCOSB-rlInterfaces::swIfOperSuspendedStatus" which has nothing to do with PoE.

If I add a debug line to my code, like this:

`            //Discover Sensors
            discover_sensor(null, 'state', $device, $cur_oid . $index, $index, $state_name, $descr, 1, 1, null, null, null, null, $entry, 'snmp', $index);
echo "DESCRIPTION: $descr\n";
`

I get:
State: DESCRIPTION: GigabitEthernet1/0/1 Suspended Status
DESCRIPTION: GigabitEthernet1/0/2 Suspended Status
DESCRIPTION: GigabitEthernet1/0/3 Suspended Status
DESCRIPTION: GigabitEthernet1/0/4 Suspended Status
DESCRIPTION: GigabitEthernet1/0/5 Suspended Status
DESCRIPTION: GigabitEthernet1/0/6 Suspended Status

My code seems fine.

Worse, the lack of interface description seems to be coming from the LibreNMS test suite?
My own LibreNMS installation generates the descriptions just fine:

`
MariaDB [librenms]> select sensor_class, sensor_oid, sensor_index, sensor_type, sensor_descr, sensor_current from sensors where device_id = 190 ;
+--------------+---------------------------------------+--------------+-------------------------+------------------------------------------+----------------+
| sensor_class | sensor_oid                            | sensor_index | sensor_type             | sensor_descr                             | sensor_current |
+--------------+---------------------------------------+--------------+-------------------------+------------------------------------------+----------------+
| power        | .1.3.6.1.4.1.9.6.1.101.108.1.1.5.1.7  | 1.7          | ciscosb                 | GigabitEthernet1/0/7 PoE                 |          6.852 |
| power        | .1.3.6.1.4.1.9.6.1.101.108.1.1.5.1.8  | 1.8          | ciscosb                 | GigabitEthernet1/0/8 PoE                 |          7.323 |
| power        | .1.3.6.1.4.1.9.6.1.101.108.1.1.5.1.9  | 1.9          | ciscosb                 | GigabitEthernet1/0/9 PoE                 |          7.148 |
| power        | .1.3.6.1.4.1.9.6.1.101.108.1.1.5.1.10 | 1.10         | ciscosb                 | GigabitEthernet1/0/10 PoE                |          7.194 |
| power        | .1.3.6.1.4.1.9.6.1.101.108.1.1.5.1.11 | 1.11         | ciscosb                 | GigabitEthernet1/0/11 PoE                |          7.133 |
| power        | .1.3.6.1.4.1.9.6.1.101.108.1.1.5.1.13 | 1.13         | ciscosb                 | GigabitEthernet1/0/13 PoE                |          9.338 |
| power        | .1.3.6.1.4.1.9.6.1.101.108.1.1.5.1.14 | 1.14         | ciscosb                 | GigabitEthernet1/0/14 PoE                |          8.719 |
| power        | .1.3.6.1.4.1.9.6.1.101.108.1.1.5.1.15 | 1.15         | ciscosb                 | GigabitEthernet1/0/15 PoE                |          8.603 |
| power        | .1.3.6.1.4.1.9.6.1.101.108.1.1.5.1.16 | 1.16         | ciscosb                 | GigabitEthernet1/0/16 PoE                |          8.985 |
`

Comment (PipoCanaja) [reply tests/data/ciscosb_cbs350-4x.json]: yes, for some reason, your test-data generation is giving funny results.

Comment (rudybroersma) [reply tests/data/ciscosb_cbs350-4x.json]: > yes, for some reason, your test-data generation is giving funny results.

Yeah, any ideas what might cause this and how to fix this? It's frustrating :-)

**Review by PipoCanaja — APPROVED** [review]
LGTM

File: tests/data/ciscosb_cbs350-4x.json (hunk 1 above)
Comment (PipoCanaja) [reply tests/data/ciscosb_cbs350-4x.json]: Do you use the `DBTEST=1` statement in your `.env` file when you generate the JSON ? 
I personnally do : 
- Add or uncomment the `DBTEST=1` statement in the .env file.
- Run `lnms dev:check -o <anyOsYouWant>` up until necessary dependencies are installed, and DB is reset, and I stop the command:
```
  import pkg_resources
Only checking os:  <anyOsYouWant>
Running unit check... 
Refreshing database...done
^C
```
- Run the `save-test-data.php` script. 

This seems to give consistant results, so far.

**Review by laf — APPROVED** [review]
LGTM

**Comment by laf** [comment]
Haha, you beat me to it @PipoCanaja

**Comment by PipoCanaja** [comment]
Thanx @rudybroersma !

**Pull request merged**